pubsublite/manager.go

```diff
@@ -240,3 +272,56 @@ func (m *Manager) DeleteSubscription(ctx context.Context, subscription string) e
 	logger.Info("deleted pubsublite subscription")
 	return nil
 }
+
+func (m *Manager) gatherMetrics(ctx context.Context, o metric.Observer) error {
+	it := m.monitoringClient.ListTimeSeries(ctx, &monitoringpb.ListTimeSeriesRequest{
+		Name:   fmt.Sprintf("projects/%s", m.cfg.Project),
+		Filter: "metric.type = \"pubsublite.googleapis.com/subscription/backlog_message_count\"",
```

Comment:
Yeah that doesn't seem ideal. Can we filter by region? https://cloud.google.com/monitoring/api/resources#tag_pubsublite_subscription_partition says that the "location" is available as a label.

We could then cross reference the returned subscription metrics with the result of `ListReservationTopics`.

I think what we should do instead though is enable the caller to pass in a list of topics. Rather than registering the metric in NewManager, we would instead provide a method which accepts a list of topics and registers a callback. Something like this:

```go
// MonitorConsumerLag registers a callback with OpenTelemetry
// to measure consumer group lag for the given topics.
func (m *Manager) MonitorConsumerLag(topics ...apmqueue.Topic) (metric.Registration, error)
```

Then we could either include those topic names as prefix filters on the subscription IDs, or filter client side.